lib/streamlit/elements/progress.py

```diff
@@ -28,8 +29,43 @@
 FloatOrInt: TypeAlias = Union[int, float]
 
 
+def _get_value(value):
+    if isinstance(value, int):
+        if 0 <= value <= 100:
+            return value
+        else:
+            raise StreamlitAPIException(
+                "Progress Value has invalid value [0, 100]: %d" % value
+            )
+
+    elif isinstance(value, float):
+        if 0.0 <= value <= 1.0:
+            return int(value * 100)
+        else:
+            raise StreamlitAPIException(
+                "Progress Value has invalid value [0.0, 1.0]: %f" % value
+            )
+    else:
+        raise StreamlitAPIException(
+            "Progress Value has invalid type: %s" % type(value).__name__
+        )
+
+
+def _get_text(text: Optional[str]) -> Optional[str]:
+    if text is None:
+        return None
+    if isinstance(text, str):
+        return clean_text(text)
+    raise StreamlitAPIException(
+        f"Progress Text is of type {str(type(text))}, which is not an accepted type."
+        "Text only accepts: str. Please convert the text to an accepted type."
+    )
+
+
 class ProgressMixin:
-    def progress(self, value: FloatOrInt) -> "DeltaGenerator":
+    def progress(
+        self, value: FloatOrInt, text: Optional[str] = None
```

Comment:
nit: since you already broke a line, maybe it would make sense to separate all arguments into separate lines?

Reply:
We use [`prettier`](https://prettier.io/) to format the code and they formatted the code that way. I don't have much control over it.